assets/js/modules/tagmanager/datastore/containers.js

```diff
@@ -0,0 +1,171 @@
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME, CONTEXT_WEB, CONTEXT_AMP } from './constants';
+import { isValidAccountID, isValidUsageContext } from '../util/validation';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const fetchGetContainersStore = createFetchStore( {
+	baseName: 'getContainers',
+	argsToParams: ( accountID ) => {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to fetch containers.' );
+
+		return { accountID };
+	},
+	controlCallback: ( { accountID } ) => {
+		// Always request both contexts to prevent filtering on server.
+		// TODO: Remove `usageContext` param when legacy component is removed and datapoint
+		// defaults to returning all containers if no context is provided.
+		const usageContext = [ CONTEXT_WEB, CONTEXT_AMP ];
+		return API.get( 'modules', 'tagmanager', 'containers', { accountID, usageContext }, { useCache: false } );
+	},
+	reducerCallback: ( state, containers, { accountID } ) => {
+		return {
+			...state,
+			containers: {
+				...state.containers,
+				[ accountID ]: [ ...containers ],
+			},
+		};
+	},
+} );
+
+const fetchCreateContainerStore = createFetchStore( {
+	baseName: 'createContainer',
+	argsToParams( accountID, usageContext ) {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to create a container.' );
+		invariant( isValidUsageContext( usageContext ), 'A valid usageContext is required to create a container.' );
+
+		return { accountID, usageContext };
+	},
+	controlCallback: ( { accountID, usageContext } ) => {
+		return API.set( 'modules', 'tagmanager', 'create-container', { accountID, usageContext } );
+	},
+	reducerCallback( state, container, { accountID } ) {
+		return {
+			...state,
+			containers: {
+				...state.containers,
+				[ accountID ]: [
+					...( state.containers[ accountID ] || [] ),
+					container,
+				],
+			},
+		};
+	},
+} );
+
+const BASE_INITIAL_STATE = {
+	containers: {},
+};
+
+const baseActions = {
+	/**
+	 * Creates a new Tag Manager container in the given account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} accountID    Google Tag Manager account ID.
+	 * @param {string} usageContext Container usage context. (Either 'web', or 'amp')
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*createContainer( accountID, usageContext ) {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to create a container.' );
+		invariant( isValidUsageContext( usageContext ), 'A valid usageContext is required to create a container.' );
+
+		const { response, error } = yield fetchCreateContainerStore.actions.fetchCreateContainer( accountID, usageContext );
+
+		return { response, error };
+	},
+};
+
+const baseResolvers = {
+	*getContainers( accountID ) {
```

Comment:
Does this work? The resolver here has different parameters from the selector.

Reply:
Yep - the resolver doesn't need to have the same signature as the selector, it is called with the same arguments though (less `state` of course).